File: ts/src/quickSort.ts
const quickSort = (arr: number[]): number[] => {
	if (arr.length < 2) return arr

	const pivotIndex = Math.floor(Math.random() * arr.length)
	const pivot = arr[pivotIndex]

	let under = []
	let above = []

	for (let [index, item] of arr.entries()) {
		if (index === pivotIndex) continue

		if (item <= pivot) under.push(item)
		else above.push(item)
	}

	return [...quickSort(under), pivot, ...quickSort(above)]
}

const array = Array.from({ length: 10 }, () => Math.ceil(Math.random() * 100))
console.log(quickSort(array))

File: py/quick_sort.py
from typing import List
import random


def quick_sort(arr: List[int]) -> List[int]:
    if len(arr) < 2:
        return arr
    else:
        pivot_index = random.randint(0, len(arr) - 1)
        pivot = arr[pivot_index]
        under, above = [], []

        for index, item in enumerate(arr):
            if pivot_index == index:
                continue
            if pivot >= item:
                under.append(item)
            else:
                above.append(item)

        return quick_sort(under) + [pivot] + quick_sort(above)


arr = random.sample(range(1, 101), 10)
print(quick_sort(arr))
